### ISSUE TITLE
[RN0.66][Android] expo-dev-launcher build fails: Using 'of(Map<String, String>): Headers' is an error. function moved to extension

### ISSUE THREAD

**Issue by Yonom** [expo/expo#16017]
### Summary

According to the [blog post](https://blog.expo.dev/expo-dev-client-0-8-0-7116c1945253) about expo-dev-client 0.8, React Native 0.66 is supported in bare workflow. However, my android build fails with the following error:

```
> Task :expo-dev-launcher:compileDebugKotlin FAILED
e: /Users/yonom/Documents/GitHub/test/yonom/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt: (46, 20): Using 'of(Map<String, String>): Headers' is an error. function moved to extension

FAILURE: Build failed with an exception.
```

### Managed or bare workflow? If you have made manual changes inside of the `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### Package versions

    "expo": "44.0.5",
    "expo-dev-client": "0.8.2",
    "react-native": "0.66.4",

### Environment

```
  Expo CLI 5.0.3 environment info:
    System:
      OS: macOS 12.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.13.2 - /opt/homebrew/opt/node@16/bin/node
      Yarn: 1.22.17 - /opt/homebrew/bin/yarn
      npm: 8.1.2 - /opt/homebrew/opt/node@16/bin/npm
      Watchman: 2022.01.03.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
      Android SDK:
        API Levels: 29, 30, 31
        Build Tools: 29.0.2, 30.0.2, 31.0.0, 32.0.0, 32.0.0
        System Images: android-29 | Google Play ARM 64 v8a
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7935034
      Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~44.0.0 => 44.0.5 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.66.4 => 0.66.4 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      eas-cli: 0.45.1
      expo-cli: 5.0.3
    Expo Workflow: bare
```

### Reproducible demo

Run the following commands inside a test directory:

- `expo init yonom --yes`
- `cd yonom`
- `expo install expo-dev-client`
- `expo prebuild`
- `yarn install react-native@0.66.4`
- `expo run:ios`

Result: :x: Build fails

**Comment by Yonom** [comment]
I was able to patch-package this issue:

`expo-dev-launcher+0.10.2.patch`
```patch
diff --git a/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt b/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt
index f34533a..0958010 100644
--- a/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt
+++ b/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt
@@ -43,6 +43,6 @@ class DevLauncherManifestParser(
     if (installationID != null) {
       headersMap["expo-dev-client-id"] = installationID
     }
-    return Headers.of(headersMap)
+    return headersMap.toHeaders()
   }
 }

```

**Comment by mariansam** [comment]
+1, happens to me too. I fixed it by manually editing the `DevLauncherManifestParser.kt` file according to your patch. Then I used [`patch-package`](https://www.npmjs.com/package/patch-package) to store and automate the patching on every run of `yarn`. This works great with Expo EAS Build.

However, my `DevLauncherManifestParser.kt` looked a bit different, I had to change
```kt
  private fun getHeaders(): Headers {
    return Headers.of(mapOf("expo-platform" to "android"))
  }

```
to
```kt
  private fun getHeaders(): Headers {
    return mapOf("expo-platform" to "android").toHeaders()
  }

```

**Comment by mark-hayward** [comment]
I did this which removed the original error but now I get : 

/node_modules/expo-dev-launcher/android/src/react-native-66/expo/modules/devlauncher/rncompatibility/DevLauncherDevSupportManager.kt: (58, 1): No value passed for parameter 'surfaceDelegateFactory'

**Comment by lukmccall** [comment]
This issue should be fixed by https://github.com/expo/expo/pull/15915. We'll try to publish that as soon as possible.

**Comment by mangkoran** [comment]
May you inform us here when the package is published?

**Comment by lukmccall** [comment]
Yes, I will ;)

**Comment by lukmccall** [comment]
@mangkoran, That fix was published in `dev-client@0.8.4`.
